trie/encoding.go

```diff
@@ -75,13 +75,16 @@ func keybytesToHex(str []byte) []byte {
 }
 
 // hexToKeybytes turns hex nibbles into key bytes.
-// This can only be used for keys of even length.
+// This is only useful for keys of even length.
 func hexToKeybytes(hex []byte) []byte {
 	if hasTerm(hex) {
 		hex = hex[:len(hex)-1]
 	}
 	if len(hex)&1 != 0 {
-		panic("can't convert hex key of odd length")
+		// Odd length. This can happen if someone iterates a Trie
```

Comment:
Isn't a 'rare production crash' the correct behaviour if we're iterating partial tries?